bgpd/bgp_updgrp_adv.c

```diff
@@ -755,37 +755,34 @@ void subgroup_default_originate(struct update_subgroup *subgrp, int withdraw)
 	}
 
 	if (peer->default_rmap[afi][safi].name) {
+		struct attr attr_tmp = attr;
+		struct bgp_path_info bpi_rmap = {0};
+
+		bpi_rmap.peer = bgp->peer_self;
+		bpi_rmap.attr = &attr_tmp;
+
 		SET_FLAG(bgp->peer_self->rmap_type, PEER_RMAP_TYPE_DEFAULT);
 		for (rn = bgp_table_top(bgp->rib[afi][safi]); rn;
 		     rn = bgp_route_next(rn)) {
```

Comment:
Just thinking aloud. This is case where there are no prefix list to match with and only set, in this case looping all the prefix just does not make any sense as all prefix will be matched anyways. Should we have a function to check if routemap really has match criteria only then enter route loop?

Reply:
I'll check regarding this.

Reply:
Actually this is not necessary because the inner loop stops right after finding the first MATCH. Thus if we have 100k routes in the RIB and only SET action then iterating over the RIB we will stop at the beginning.

Reply:
Right below code should take care of them same.

if (ret != RMAP_DENYMATCH)
				break;